test/gtest/uct/test_md.cc

```diff
@@ -743,15 +746,14 @@ static void test_alloc_advise(uct_md_h md_ref, ucs_memory_type_t mem_type)
 UCS_TEST_SKIP_COND_P(test_md, alloc_advise,
                      !check_caps(UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_ADVISE))
 {
-    uct_md_h md_ref = md();
+    uct_md_h md_ref    = md();
+    uint64_t mem_types = md_attr().alloc_mem_types & md_attr().access_mem_types;
     ucs_memory_type_t mem_type;
     uint32_t mem_type_index;
 
-    ucs_for_each_bit(mem_type_index, md_attr().alloc_mem_types) {
-        if (md_attr().access_mem_types & UCS_BIT(mem_type_index)) {
-            mem_type = static_cast<ucs_memory_type_t>(mem_type_index);
-            test_alloc_advise(md_ref, mem_type);
-        }
+    ucs_for_each_bit(mem_type_index, mem_types) {
+        mem_type = static_cast<ucs_memory_type_t>(mem_type_index);
```

Comment:
why need both mem_type and mem_type_index variables?
only mem_type should be enough

Reply:
CPP won't accept treating ucs_memory_type_t as uint32_t, I can declare `mem_type` as `uint32_t` and then `static_cast` inline if that what you meant

Reply:
yeah, this looks better. but do we really need the static_cast? won't it happen automatically if we just pass mem_type?

Reply:
we do need the cast

Reply:
ok